### PR TITLE
Add differenceWith

### PR DESCRIPTION
**Pull request by vlad-yakovlev**
Add the same function as presented in [Lodash](https://lodash.com/docs/4.17.15#differenceWith) and [Ramda](https://ramdajs.com/docs/#differenceWith)

Diff so big because of changes in `mapping.md`. Code and tests mostly the same as in the `difference` method

### CODE REVIEW

File: src/differenceWith.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+/**
+ * Excludes the values from `other` array.
+ * @param array the source array
+ * @param other the values to exclude
+ * @param isEquals the comparator
+ * @signature
+ *    R.differenceWith(array, other, isEquals)
+ * @example
+ *    R.differenceWith([1, 2, 3, 4], [2, 5, 3], R.equals) // => [1, 4]
```
Comment (TkDodo) [inline src/differenceWith.ts:14]: same here, lets please use better examples & tests

Comment (vlad-yakovlev) [reply src/differenceWith.ts]: Fixed

**Review by TkDodo — APPROVED** [review]

**Comment by TkDodo** [comment]
ah, we missed adding the export to `index.ts` 🤦 . Can you add that please?

**Comment by vlad-yakovlev** [comment]
Yep, 2 min and I'll open new PR :)

**Pull request merged**